tests/artifact/test_local_artifact.py

```diff
@@ -0,0 +1,254 @@
+    assert result is None
+
+
+def test_find_artifact_paths_from_invalid_python_venv() -> None:
+    """Test find_artifact_paths_from_python_venv method with invalid venv path"""
+    assert not find_artifact_paths_from_python_venv("./does-not-exist", ["django", "django-5.0.6.dist-info"])
+
+
+@pytest.mark.parametrize(
+    ("purl_str", "build_tool_purl_types", "local_artifact_repo_mapper", "expectation"),
+    [
+        pytest.param(
+            "pkg:maven/com.google.guava/guava@33.2.1-jre",
+            ["maven", "pypi"],
+            {},
+            {},
+            id="A maven type PURL where multiple build tool types are discovered. But no local repository is available.",
+        ),
+        pytest.param(
+            "pkg:maven/com.google.guava/guava@33.2.1-jre",
+            [],
+            {},
+            {},
+            id="A maven type PURL where no build tool types are discovered and no local repository is available.",
+        ),
+    ],
+)
+def test_get_local_artifact_paths_empty(
+    purl_str: str,
+    build_tool_purl_types: list[str],
+    local_artifact_repo_mapper: dict[str, str],
+    expectation: dict[str, list[str]],
+) -> None:
+    """Test getting local artifact paths where the result is empty."""
+    purl = PackageURL.from_string(purl_str)
+    assert (
+        get_local_artifact_paths(
+            purl=purl,
+            build_tool_purl_types=build_tool_purl_types,
+            local_artifact_repo_mapper=local_artifact_repo_mapper,
+        )
+        == expectation
+    )
+
+
+@pytest.mark.parametrize(
+    ("purl_str", "build_tool_purl_types", "expectation"),
+    [
+        pytest.param(
+            "pkg:maven/com.google.guava/guava@33.2.1-jre",
+            ["maven", "pypi"],
+            {},
+            id="A maven type PURL where multiple build tool types are discovered",
+        ),
+        pytest.param(
+            "pkg:maven/com.google.guava/guava@33.2.1-jre",
+            [],
+            {},
+            id="A maven type PURL where no build tool is discovered",
+        ),
+        pytest.param(
+            "pkg:pypi/django@5.0.3",
+            [],
+            {},
+            id="A maven type PURL where no build tool is discovered",
+        ),
+    ],
+)
+def test_get_local_artifact_paths_not_available(
+    purl_str: str,
+    build_tool_purl_types: list[str],
+    expectation: dict[str, list[str]],
+    tmp_path: Path,
+) -> None:
+    """Test getting local artifact paths where the artifact paths are not available."""
+    purl = PackageURL.from_string(purl_str)
+    local_artifact_repo_mapper = {
+        "maven": str(tmp_path),
+        "pypi": str(tmp_path),
+    }
+
+    assert (
+        get_local_artifact_paths(
+            purl=purl,
+            build_tool_purl_types=build_tool_purl_types,
+            local_artifact_repo_mapper=local_artifact_repo_mapper,
+        )
+        == expectation
+    )
+
+
+def test_get_local_artifact_paths_succeeded_maven(tmp_path: Path) -> None:
```

Comment:
Why are PyPi related things included in this function? And the same question for Maven related things in the next function.

Reply:
This answer also applies to `test_get_local_artifact_paths_succeeded_pypi`.
The reason for having both of those bits originated from my implementation. I intended to support cases where for a single PURL, we could detect multiple build tools from its source code, hence multiple build tool PURL types. This follows what we have here - https://github.com/oracle/macaron/blob/4235041f706c7a811b5ddf369e26549bdea0206f/src/macaron/slsa_analyzer/analyzer.py#L885 where for a single target we could have multiple build tool purl types.
So my implementation ensures that for any PURL, it will be checked against all local repositories corresponding to build tools Macaron discovered for the PURL's source code. However, as we currently only support the type of input PURL to be either "maven" or "pypi", a PURL can only map to one type of local artifact ("maven" or "pypi" or not supported). 

My original intention was to make this as generic as possible. If it's a bit confusing, I can remove this for now and add them back if we think it's necessary 🤔

Reply:
I see. I would still vote for not including them currently then, as my understanding is that they have no effect on the test behaviour.

Reply:
I think the unit tests will need to be changed a bit. 
This is because if we remove the behavior of looking for artifacts in each build tool purl type, and only use the type of the input PURL, then the interface of `get_local_artifact_paths` will become
```
def get_local_artifact_paths(
    purl: PackageURL,
    local_artifact_repo_path: str,
) -> list[str]:
```

Reply:
Addressed in https://github.com/oracle/macaron/pull/864/commits/7adbee2cbd4ee972af55aefc50f7acd4fe6214c2